### PR TITLE
The graph subdomains and address => name list

### PR DESCRIPTION
**Pull request by jefflau**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1 +1,2 @@
-NODE_PATH=src/
\ No newline at end of file
+NODE_PATH=src/
+REACT_APP_GRAPH_NODE_URI=http://localhost:8000/subgraphs/name/graphprotocol/ens
```
Comment (makoto) [inline .env]: Please add new line

File: README.md [hunk 2]
```diff
@@ -119,3 +119,77 @@ yarn start
...
+
+Download and start [docker](https://www.docker.com/) first
+
+### Start thegraph node
+
+This starts up docker with ipfs, postgresdb, and the-graph node.
+
+```
+cd ../graph-node
+docker-compose up
```
Comment (makoto) [inline README.md]: There was a mistake in my doc. Please change to the below
```
cd ../graph-node
cd docker
docker-compose up
```

File: scripts/subgraph.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env node
+yaml = require('js-yaml')
+fs = require('fs')
+var fileName = '../ens-subgraph/subgraph.yaml'
+var doc = yaml.safeLoad(fs.readFileSync(fileName))
+var addresses = JSON.parse(fs.readFileSync('./cypress.env.json', 'utf8'))
+
+console.log(doc, addresses)
```
Comment (makoto) [inline scripts/subgraph.js]: Forgot to remove this. Can you remove the console.log?

File: src/api/labels.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+  if (hash.startsWith('0x')) {
+    return labels[`${hash.slice(2)}`]
+  }
+}
+
+export function decryptName(name) {
+  return name
+    .split('.')
+    .map(label => {
```
Comment (makoto) [inline src/api/labels.js]: Can be done in one liner?
```
 .map(label => { return checkLabel(label) || label }
```

File: src/api/labels.test.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
+import { checkLabel, saveLabel, saveName } from './labels'
+const KEY = 'labels'
+
+beforeEach(() => {
+  localStorage.clear()
+})
+
+function setupStorage() {
+  localStorage.__STORE__[KEY] = JSON.stringify({
+    '0x36940f34a2ec6afe46b7db53e6611470cd76c4f5999209a04a670682e2c33f75':
```
Comment (makoto) [inline src/api/labels.test.js]: Do you want to put these as constant at the top of the test file?
It's a bit hard to see which hash are identical and which is not. Same for `blahblah` (you can put into `label` constant or something).

File: src/api/labels.test.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    ).toBe('blahblah')
+  })
+
+  test('should return undefined if label is not in localStorage', () => {
+    setupStorage()
+    const hash =
+      '0x36940f34a2ec6afe46b7db53e6611470cd76c4f5999209a04a670682e2c33f75'
+    expect(
+      checkLabel(
+        '0x36940f34a2ec6afe46b7db53e6611470cd76c4f5999209a04a670682e2c33f74'
```
Comment (makoto) [inline src/api/labels.test.js]: I can only figure out that this hash is different to the one at line 34, by looking at the last digit. Please turn it into constant/variable so easier to know they are different

File: src/apolloClient.js [hunk 7]
```diff
@@ -1,23 +1,75 @@
...
+}
+
+function getGraphQLAPI(network) {
+  if (endpoints[network]) {
+    return endpoints[network]
+  } else if (process.env.REACT_APP_GRAPH_NODE_URI) {
+    return process.env.REACT_APP_GRAPH_NODE_URI
+  } else {
+    return endpoints['1']
+  }
```
Comment (makoto) [inline src/apolloClient.js]: `return endpoints[network] || process.env.REACT_APP_GRAPH_NODE_URI || endpoints['1']`

**Comment by makoto** [comment]
When I query subdomain of the domain with lots of subdomain (eg: http://localhost:3000/name/eth/subdomains), it just shows no result. Probably it should either offer pagination or at least "it's too many" error message.

https://gist.github.com/makoto/62d644997a3a6565355593bd59592e5c

**Comment by jefflau** [comment]
> When I query subdomain of the domain with lots of subdomain (eg: http://localhost:3000/name/eth/subdomains), it just shows no result with [some errors on console](https://gist.github.com/makoto/62d644997a3a6565355593bd59592e5c). Probably it should either offer pagination or at least "it's too many" error message.

Good idea. I'll add that in

**Comment by makoto** [comment]
There is no indication on the page that you can search by address. Maybe "Search name" should be renamed by "Search ENS name or hex address"?

**Pull request merged**